graviton/blackbox.py

```diff
@@ -174,406 +196,385 @@ def _partial_encode_assert(
         """
-        params = dict(
-            sender=sender,
-            sp=sp,
-            note=note,
-            lease=lease,
-            rekey_to=rekey_to,
-            receiver=receiver,
-            amt=amt,
-            close_remainder_to=close_remainder_to,
-            index=index,
-            on_complete=on_complete,
-            local_schema=local_schema,
-            global_schema=global_schema,
-            approval_program=approval_program,
-            clear_program=clear_program,
-            app_args=app_args,
-            accounts=accounts,
-            foreign_apps=foreign_apps,
-            foreign_assets=foreign_assets,
-            extra_pages=extra_pages,
+        executor = self._executor(txn_params, verbose)
+        if isinstance(inputs, tuple):
+            return executor(inputs)
+
+        assert isinstance(
+            inputs, (list, map)
+        ), f"inputs must be of type list or map (for multiple args) or tuple (for single args) but was {type(inputs)}"
+        if isinstance(inputs, map):
+            inputs = list(inputs)
+
+        assert inputs, "must provide at least one input args tuple"
+
+        for i, args in enumerate(inputs):
+            assert isinstance(
+                args, tuple
+            ), f"each args in inputs list must be a tuple but at index {i=} we have {type(args)}"
+
+        inputs = cast(List[Tuple[PyTypes, ...]], inputs)
+        return list(map(executor, inputs))
+
+    def _executor(
+        self,
+        txn_params: Optional[DryRunTransactionParams],
+        verbose: bool,
+    ) -> Callable[[Tuple[PyTypes, ...]], DryRunInspector]:
+        def executor(args: Tuple[PyTypes, ...]) -> DryRunInspector:
+            args, encoded_args = self._executor_prep(args)
+
+            dryrun_req: DryrunRequest
+            txn_params_d = txn_params.asdict() if txn_params else {}
+            dr_acts = txn_params.dryrun_accounts if txn_params else []
+            if self.is_app:
+                dryrun_req = DryRunHelper.singleton_app_request(
+                    self.program,
+                    encoded_args,
+                    txn_params_d,
+                    dr_acts,
+                )
+            else:
+                dryrun_req = DryRunHelper.singleton_logicsig_request(
+                    self.program, encoded_args, txn_params_d
+                )
+            if verbose:
+                print(f"{type(self)}._run(): {dryrun_req=}")
+            dryrun_resp = self.algod.dryrun(dryrun_req)
+            if verbose:
+                print(f"{type(self)}::_executor(): {dryrun_resp=}")
+            return DryRunInspector.from_single_response(
+                dryrun_resp, args, encoded_args, abi_type=self.abi_return_type
+            )
+
+        return executor
+
+    def _executor_prep(
+        self, args: Tuple[PyTypes, ...]
+    ) -> Tuple[Tuple[PyTypes, ...], List[ArgType]]:
+        """ """
```

Comment:
Empty comments, I guess it can go away?

Reply:
see df452d53d4e5e1335708489b0dfb1ad9e807a12c

Reply:
lgtm